pkg/controller/rhmiconfig/helpers/update_status.go

```diff
@@ -0,0 +1,101 @@
+package helpers
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const WINDOW = 6
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integreatlyv1alpha1.RHMIConfig, targetVersion string) error {
+	// Calculate the next maintenance window based on the maintenance schedule
+	if config.Spec.Maintenance.ApplyFrom != "" {
+		mtStart, _, err := getWeeklyWindowFromNow(config.Spec.Maintenance.ApplyFrom, time.Hour*WINDOW)
+		if err != nil {
+			return err
+		}
+
+		config.Status.Maintenance.ApplyFrom = mtStart.Format("2-1-2006 15:04")
+		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
+	}
+
+	config.Status.UpgradeAvailable.TargetVersion = targetVersion
+
+	client.Status().Update(ctx, config)
```

Comment:
```suggestion
```

This can go as it's meant to be set by the subscription controller to trigger a reconciliation of the RHMIConfig controller on change.